### ISSUE TITLE
Why does ProjectedGradientDescent use CallableModelWrapper with "probs"?

### ISSUE THREAD

**Issue by goodfeli** [tensorflow/cleverhans#744]
The __init__ method of ProjectedGradientDescent will accept a callable rather than a Model and wraps it in a CallableModelWrapper that names the output "probs".

This makes me nervous. It seems like only one of two things can be happening:
1) This code path is just broken, and when you try to run the attack you get a NotImplementedError telling you you need to provide the logits. This isn't dangerous, but it would be better to just raise a TypeError in the constructor and say to provide a Model class.
2) Less likely, but I'm worried maybe we end up using probs as logits somehow, in which case the gradient-based optimizer probably performs badly.
